kernprof.py

```diff
@@ -307,12 +307,27 @@ def stop(self):
         self.is_running = False
 
 
-def find_module_script(module_name, *, exit_on_error=True):
+def find_module_script(module_name, *, static=True, exit_on_error=True):
     """Find the path to the executable script for a module or package."""
     from line_profiler.autoprofile.util_static import modname_to_modpath
+    from importlib.util import find_spec
+
+    @_restore.mapping(sys.modules)
+    def resolve_module_path(mod_name):  # type: (str) -> str | None
```

Comment:
Are you doing this restore to avoid the side effect of having the module in `sys.modules`? The only reason I had an issue with that is because I like to avoid arbitrary code execution and to improve response time. We've already paid the price here, so unless it is important to restore sys.modules for a different reason, we can leave it populated.